docs/db/db-relations.md

```diff
@@ -0,0 +1,191 @@
+
+Example:
+
+Campus `madeira` is a **parent object** of `piscine-go`, therefore the later one is the **child**.
+
+But `piscine-go` can be the **parent object** of all the quest, exams and raids (those being the **child objects**). And so on... creating a finite cycle.
+
+---
+
+### Users
+
+Users will be the accumulation of the following tables:
+
+- `role`
+- `user_role`
+- `user`
+- `group`
+- `group_user`
+- `token`
+- `record`
+- `transaction`
+
+---
+
+The `role` table contains permission roles for each user.
+
+---
+
+The `user_role` contains information on which users are associated to which role. A user can have multiple roles and a role can be associated to multiple users.
+
+---
+
+The `group` table is the link between projects and a group of users. This table contains the following relations:
+
+- `eventId` that associates which group is related to which event.
+- `captainId` that associates the captain's user.
+- `objectId` that associates which group is related to which object.
+
+---
+
+The `group_user` table contains the relation between groups and users. A group can have several users and so do the users.
+
+---
+
+The `token` table stores the tokens ids from the hasura authorization variables for each user. This table has no relation between other tables.
+
+---
+
+The `record` table takes care of students records (bans). All relations in this table are with the table user.
+
+---
+
+The `transaction` table takes care of rewarding the user, by accumulating the user's **xp**. This table contains the following relations:
+
+- `userId` that represents the user rewarded
+- `eventId` that associates the event in which the user was rewarded. Example: the user can be rewarded for an exercise `/madere/piscine-go/quest-01/make-it-better` and the parent event of that exercise will be `/madere/piscine-go`.
+- `objectId` that associates the object in which the user was rewarded. Example: if a user passes an exercise (object), the reward will be associated to that object.
+
+---
+
+#### Results
+
+Results will be the accumulation of the following tables:
+
+- `audit`
+- `match`
+- `progress`
+- `result`
+
+This part of the database defines the users/students progress in the school.
+
+The `audit` table contains all information related to the audit system and it is one of the ways of obtaining results. This table contains the following relations:
+
+- `groupId` that associates the group being audited.
+- `auditorId` as the name says, it is linked to the user table and has a relation of many to one. This column will be the auditor.
+- `resultId` that associates the audit to the result. An audit can have one or no results (pending on the auditor review). While the results must have at least one audit.
+
+---
+
+The `match` table is another way of obtaining a result. This table is used in bonus exercises to match two students. The following relations are established:
```

Comment:
there is actually no relation between the tables match and result in db right now, but there should be after the refactor of the tester! :)

Reply:
yes, in the meeting it was clear that it will have that relation in the near future.